pyccel/codegen/printing/fcode.py

```diff
@@ -497,6 +506,45 @@ def _calculate_class_names(self, expr):
                         scope.rename_function(f, suggested_name)
                     f.cls_name = scope.get_new_name(f'{name}_{f.name}')
 
+    def _build_gFTL_module(self, expr_type):
+        """
+        Build the gFTL module to create container types.
+
+        Create a module which will import the gFTL include files
+        in order to create container types (e.g lists, sets, etc).
+        The name of the module is derived from the name of the type.
+
+        Parameters
+        ----------
+        expr_type : DataType
+            The data type to be defined in a gFTL module.
+
+        Returns
+        -------
+        Import
+            The import which allows the new type to be accessed.
+        """
+        if expr_type in self._generated_gFTL_extensions:
+            module = self._generated_gFTL_extensions[expr_type]
+            mod_name = module.name
+        else:
+            if isinstance(expr_type, HomogeneousListType):
+                include = Import(LiteralString('vector/template.inc'), Module('_', (), ()))
+                macros = [MacroDefinition('T', expr_type.element_type),
+                          MacroDefinition('Vector', expr_type),
+                          MacroDefinition('VectorIterator', expr_type, '_Iterator')]
```

Comment:
@jalalium I'm not sure that this approach with the suffix is the right one. Maybe I should create a dedicated datatype. What do you think?

Reply:
I am not sure what dedicated datatype you have in mind. After further investigation, the only concern i have with the suffix approach is that, sometimes, we might need to construct macros with prefixes as well, like in the case of the const prefix.

Reply:
For example in c_concepts.py
```python
class Iterator:
    def __init__(self, iterable_type):
       ...
```
So a `List_int_iterator` would be `Iterator(HomogeneousListType(NativeInteger()))`

I suspect that this will be more general as we probably shouldn't be declaring macros for things that we won't be using elsewhere.

When I wrote this I had in mind that the iterator would only be used locally as a temporary but thinking about it later I wondered if it will still need declaring explicitly with the other variables. If this is even potentially the case then I think suffix is probably a bad habit.

For the const prefix we should be able to use the existing `Final` class

Reply:
Alternatively a macro could be defined with 2 strings instead of a string, an object and corrections. But maybe this would encourage lazy usage?

Reply:
The more we discuss this, the more I think a dedicated type is better. I have made the necessary changes